### PR TITLE
Added nuxt script snippets

### PR DESCRIPTION
**Pull request by DominusKelvin**
Added Nuxt Pages related code snippets for options like asyncData, Fetch etc.

### CODE REVIEW

File: snippets/nuxt-script.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "Nuxt Async": {
+    "prefix": "nasyncdata",
+    "body": ["async asyncData ({ ${1:params} }) {", "\tconst { data } = await axios.get(`${2:endpoint}`)", "\treturn { ${3:key}:${4:value} }", "},"],
+    "description": "Nuxt asyncData"
+  },
+  "Nuxt Async Promise": {
+    "prefix": "nasyncdata-p",
+    "body": ["asyncData ({ ${1:params} }) {", "\treturn axios.get(`${2:endpoint}`)", "\t\t.then((res) => {", "\t\t\t\treturn { ${3:key}:${4:res.data} }", "\t\t})", "},"],
```
Comment (Atinux) [inline snippets/nuxt-script.json]: I think we can remove this one to recommend the `const { data }` syntax.

File: snippets/nuxt-script.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Nuxt asyncData - Returning a Promise"
+  },
+  "Nuxt Fetch": {
+    "prefix": "nfetch",
+    "body": ["async fetch ({ store, ${1:params} }) {", "\tlet { data } = await axios.get('${2:endpoint}')", "\tstore.commit('${3:MUTATION_TYPE}', data)", "},"],
+    "description": "Nuxt Fetch"
+  },
+  "Nuxt Fetch Promise": {
+    "prefix": "nfetch-p",
+    "body": ["fetch ({ store, ${1:params} }) {", "\treturn axios.get('${2:endpoint}')", "\t.then((res) => {", "\t\tstore.commit('${3:MUTATION_TYPE}', ${4:res.data})", "\t})","},"],
```
Comment (Atinux) [inline snippets/nuxt-script.json]: I think we can remove this one to recommend the `const { data }` syntax.

File: snippets/nuxt-script.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "Nuxt Async": {
+    "prefix": "nasyncdata",
+    "body": ["async asyncData ({ ${1:params} }) {", "\tconst { data } = await axios.get(`${2:endpoint}`)", "\treturn { ${3:key}:${4:value} }", "},"],
```
Comment (Atinux) [inline snippets/nuxt-script.json]:
```suggestion
    "body": ["async asyncData ({ ${1:params} }) {", "\tconst { data } = await fetch(`${2:endpoint}`).then(res => res.json())", "\treturn { ${3:key}:${4:value} }", "},"],
```

Using `fetch` will allow to remove `axios` dependency :)

File: snippets/nuxt-script.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    "description": "Nuxt asyncData"
+  },
+  "Nuxt Async Promise": {
+    "prefix": "nasyncdata-p",
+    "body": ["asyncData ({ ${1:params} }) {", "\treturn axios.get(`${2:endpoint}`)", "\t\t.then((res) => {", "\t\t\t\treturn { ${3:key}:${4:res.data} }", "\t\t})", "},"],
+    "description": "Nuxt asyncData - Returning a Promise"
+  },
+  "Nuxt Fetch": {
+    "prefix": "nfetch",
+    "body": ["async fetch ({ store, ${1:params} }) {", "\tlet { data } = await axios.get('${2:endpoint}')", "\tstore.commit('${3:MUTATION_TYPE}', data)", "},"],
```
Comment (Atinux) [inline snippets/nuxt-script.json]:
```suggestion
    "body": ["async fetch ({ store, ${1:params} }) {", "\tlet { data } = await fetch('${2:endpoint}').then(res => res.json())", "\tstore.commit('${3:MUTATION_TYPE}', data)", "},"],
```

Same here to remove axios dependency :)

**Comment by DominusKelvin** [comment]
Thanks, Sébastien for reviewing. :love:

**Comment by DominusKelvin** [comment]
@sdras. What do you think of the additions I made?

**Comment by sdras** [comment]
Wonderful, thank you!

**Pull request merged**